test/test-parse-function.cpp

```diff
@@ -851,6 +851,37 @@ TEST(test_parse, unnamed_function_expression_without_body) {
   }
 }
 
+TEST(test_parse, arrow_function_invoked_with_parens) {
+  {
+    padded_string code(u8"(() => {})()"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    p.parse_and_visit_module(v);
+    EXPECT_THAT(v.visits, ElementsAre("visit_enter_function_scope",
+                                      "visit_enter_function_scope_body",
+                                      "visit_exit_function_scope",
+                                      "visit_end_of_module"));
+    EXPECT_EQ(v.errors.size(), 0);
```

Comment:
Nit: The following will print the errors if the assertion fails:
```suggestion
    EXPECT_THAT(v.errors, IsEmpty());
```